Microsoft.Research/Contracts/MsCorlib/System.Math.cs

```diff
@@ -93,29 +93,40 @@ public static Decimal Abs(Decimal value)
     [Pure]
     public static float Abs(float value)
     {
+      Contract.Ensures((Contract.Result<float>() >= 0.0) || float.IsNaN(Contract.Result<float>()) || float.IsPositiveInfinity(Contract.Result<float>()));
```

Comment:
:question: Since `float.PositiveInfinity >= 0.0f`, why would we need to include it as a separate item in this condition?